backend/api/student/simulation_instances.py

```
@@ -193,3 +871,267 @@ async def complete_simulation_instance(
+                "hints_used": instance.hints_used,
+                "is_overdue": instance.is_overdue,
+                "days_late": instance.days_late,
+                "created_at": instance.created_at,
+                "updated_at": instance.updated_at
+            })
+        
+        logger.info(f"Successfully retrieved {len(result)} student instances for assignment {assignment_id}")
+        return result
+        
+    except Exception as e:
+        logger.error(f"Error fetching student instances for assignment {assignment_id}: {str(e)}", exc_info=True)
```

Ruff (0.13.3)

Use explicit conversion flag

Replace with conversion flag

(RUF010)